src/vulkan-renderer/render_graph.cpp

```diff
@@ -512,18 +512,17 @@ void RenderGraph::compile(const RenderResource *target) {
                 }
 
                 std::vector<VkImageView> image_views;
-                for (std::uint32_t i = 0; i < m_swapchain.image_count(); i++) {
-                    image_views.clear();
-                    for (const auto *back_buffer : back_buffers) {
-                        image_views.push_back(back_buffer->m_swapchain.image_view(i));
+                image_views.reserve(back_buffers.size() + images.size());
+                for (auto *const img_view : m_swapchain.image_views()) {
+                    for (std::size_t i = 0; i < back_buffers.size(); i++) {
+                        image_views.push_back(img_view);
```

Comment:
```suggestion
                    for (std::size_t idx = 0; idx < back_buffers.size(); idx++) {
                        image_views.push_back(img_view);
```

Reply:
let's use `img_index`.

Reply:
Wait. There's another issue: What I'm doing here essentially is to mimic the standard library, which is bad. I want to insert the value `img_view` into a vector `n` times. That's what we have `std::fill_n` for: https://stackoverflow.com/questions/30998444/add-same-value-multiple-times-to-stdvector-repeat

The new code looks like this:

```cpp
std::vector<VkImageView> image_views;
image_views.reserve(back_buffers.size() + images.size());
for (auto *const img_view : m_swapchain.image_views()) {
    std::fill_n(std::back_inserter(image_views), back_buffers.size(), img_view);
    for (const auto *image : images) {
        image_views.push_back(image->m_image_view);
    }
    physical.m_framebuffers.emplace_back(m_device, physical.m_render_pass, image_views, m_swapchain,
                                         "Framebuffer");
    image_views.clear();
}

```

Done.